src/Illuminate/Database/Eloquent/Relations/Concerns/CanBeOneOfMany.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Relations\Concerns;
+
+use Closure;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use InvalidArgumentException;
+
+trait CanBeOneOfMany
+{
+    /**
+     * Determines whether the relationship is one-of-many.
+     *
+     * @var bool
+     */
+    protected $isOneOfMany = false;
+
+    /**
+     * The name of the relationship.
+     *
+     * @var string
+     */
+    protected $relationName;
+
+    /**
+     * Add constraints for inner join subselect for one of many relationships.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  string|null  $column
+     * @param  string|null  $aggregate
+     * @return void
+     */
+    abstract public function addOneOfManySubQueryConstraints(Builder $query, $column = null, $aggregate = null);
+
+    /**
+     * Get the columns the determine the relationship groups.
+     *
+     * @return array|string
+     */
+    abstract public function getOneOfManySubQuerySelectColumns();
+
+    /**
+     * Add join query constraints for one of many relationships.
+     *
+     * @param  \Illuminate\Database\Eloquent\JoinClause  $join
+     * @return void
+     */
+    abstract public function addOneOfManyJoinSubQueryConstraints(JoinClause $join);
+
+    /**
+     * Indicate that the relation is a single result of a larger one-to-many relationship.
+     *
+     * @param  string|array|null  $column
+     * @param  string|Closure|null  $aggregate
+     * @param  string|null  $relation
+     * @return $this
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function ofMany($column = 'id', $aggregate = 'MAX', $relation = null)
+    {
+        $this->isOneOfMany = true;
+
+        $this->relationName = $relation ?: $this->guessRelationship();
+
+        $keyName = $this->query->getModel()->getKeyName();
+
+        $columns = is_string($columns = $column) ? [
+            $column => $aggregate,
+            $keyName => $aggregate,
+        ] : $column;
+
+        if (! array_key_exists($keyName, $columns)) {
+            $columns[$keyName] = 'MAX';
+        }
```

Comment:
Is this needed for proper functionality? It doesn't allow this feature to be used with UUID columns.